linode/manager
# change: [M3-7802] - Remove API caching

## Description 📝

- Removes all API caching from Cloud Manager
- Creates `Kernel` factory
- Updates `Kernel` type to match what the API actually returns
- Fixes a handful of tests that were built under the assumption that regions would always be loaded into memory

## Why ❓ 
- Caching large amounts of API data increases our bundle size 📦
- The API has gotten faster 💨
- Allows you to run `yarn build` without an internet connection because we no longer need to fetch from the API 🛜
- Simplifies things because there is one less moving part to worry about and maintain 🧹

## What will be removed and why ℹ️ 
- `src/cachedData/typesLegacy.json` because it was unused
- `src/cachedData/kernels.json` because it was only used for unit testing (I replaced it with a factory 🏭)
- `src/cachedData/marketplace.json` because the API now responds much faster (~300ms) (use to take many seconds ⏲️ )
- `src/cachedData/regions.json` because it was an over-optimization, not gitignored, and was misused as mock data - see https://github.com/linode/manager/pull/10191#discussion_r1488642973 (this was recently corrected by @abailly-akamai but I don't want it to happen again in the future) 🚓

## How to test 🧪

- Check out this PR or use the preview link
- Verify region data still populates correctly throughout the app
- Verify marketplace apps load in a reasonable amount of time
- Verify Kernel related features still work (this is found in a Linode's configuration)
- Verify that all automated checks pass

## As an Author I have considered 🤔
